stan/math/rev/mat/fun/cholesky_decompose.hpp

```diff
@@ -46,81 +53,102 @@ namespace stan {
                     (A.rows() * (A.rows() + 1) / 2)),
           variRefL_(ChainableStack::memalloc_.alloc_array<vari*>
                     (A.rows() * (A.rows() + 1) / 2)) {
-        size_t accum = 0;
-        size_t accum_i = accum;
-        for (size_type j = 0; j < M_; ++j) {
-          for (size_type i = j; i < M_; ++i) {
-            accum_i += i;
-            size_t pos = j + accum_i;
-            variRefA_[pos] = A.coeffRef(i, j).vi_;
-            variRefL_[pos] = new vari(L_A.coeffRef(i, j), false);
+            size_t pos = 0;
+            block_size_ = M_/8;
+            block_size_ = (block_size_/16)*16;
+            block_size_ = (std::min)((std::max)(block_size_, 8), 128);
```

Comment:
You can make `block_size_` constant and compute it at construction time in one expression that replaces `block_size_` in the last step with `((block_size_ / 8 / 16) * 16)`  [those inner parens are unnecessary but everyone seems to like them---the outer ones may be required if it's a macro]